core/crypto/peer_impl.go

```diff
@@ -69,7 +69,27 @@ func (peer *peerImpl) TransactionPreValidation(tx *obc.Transaction) (*obc.Transa
 			return tx, err
 		}
 
-		// TODO: verify cert
+		// Verify transaction certificate against root
+		// DER to x509
+		x509Cert, err := primitives.DERToX509Certificate(tx.Cert)
+		if err != nil {
+			peer.Debugf("Failed parsing certificate [% x]: [%s].", tx.Cert, err)
+
+			return tx, err
+		}
```

Comment:
@JonathanLevi, thanks for comment. Actually, I agree, the function can be converted to a predicate. Let's have this addressed by another issue. Is it okay?